### DISCUSSION TITLE
Stopping camera from following a target

### DISCUSSION THREAD

**Discussion by jjolmo** [ramokz/phantom-camera#342] [Question / Help Wanted]
Hi! I want to know if there's some exposed variable to stop the camera from following a node, at least temporarily.

I just found that should_follow = false will do the trick, at least in the play editor. But I haven't found out how to change it via code.
I'm using Godot c# so this is a bit more complex to interact but finally, it should work like this:

```
        phantomCam.Call("should_follow", false);
```

which is a similar way to do other functions like this:
```
        phantomCam.Call("set_follow_target", player);
```

Sadly, calling should_follow gives error "Invalid Call. Nonexistent function "should_follow" in base Godot.Node2D"

So. How can we prevent the camera to follow a target via code?

Thanks!

**Comment by jjolmo** [comment]
Oh. I'm sorry. I just saw that we need to use .Set, not .Call in c#

**Comment by jjolmo** [comment]
Reopening. Doesn't seem to do anything:
```
phantomCam.Set("should_follow", false);
```

Is there some way to just stop the camera from following a node?

**Reply by ramokz** [reply]
`should_follow` is only used internally within the system to check whether a camera has a `follow_target` or not. Modifying that property directly can lead to unexpected results, so wouldn't recommend messing with that one.

If you want the camera, or rather `PCam`, ~from doing any logic at all, including `Look At` behaviour~ (missed that you had specified it was a 2D scene, so this is comment is irrelevant), then a simple answer would be to disable its `process` with `pcam.process_mode = Node.PROCESS_MODE_DISABLED`. Conversely, to re-enable it, you can set the `pcam.process_mode = Node.PROCESS_MODE_INHERIT`.

**Comment by jjolmo** [comment]
Hmm.. I managed to solve the issue by creating two cameras and alter the priority like this:
```
  public void StopFollowing() {
        cameraStopped = true;
        phantomCamFixed.GlobalPosition = Game.server.room.GetNode<Node2D>("PhantomCamera2D").GlobalPosition;
        phantomCam.Call("set_priority", 0);
        phantomCamFixed.Call("set_priority", 2);
    }
```

but I this would not solve the question of just stopping a single camera
